libuserhelper.cpp

```diff
@@ -84,23 +87,64 @@ bool LibUserHelper::addUserToGroup(const QString &user, const QString &group)
     struct lu_error *error = nullptr;
     struct lu_context *context = lu_start(NULL, lu_user, NULL, NULL, NULL, NULL, &error);
     if (!context) {
-        qCWarning(lcSUM) << "Error creating context";
+        qCWarning(lcSUM) << "Error creating context:" << lu_strerror(error);
+        lu_error_free(&error);
         return false;
     }
 
     bool rv = true;
     struct lu_ent *ent = lu_ent_new();
-    if (lu_group_lookup_name(context, group.toLocal8Bit(), ent, &error)) {
+    if (lu_group_lookup_name(context, group.toUtf8(), ent, &error)) {
         GValue *value = g_slice_new0(GValue);
         g_value_init(value, G_TYPE_STRING);
         g_value_set_string(value, user.toUtf8());
         lu_ent_add(ent, LU_MEMBERNAME, value);
         if (!lu_group_modify(context, ent, &error)) {
             qCWarning(lcSUM) << "Group modify failed:" << lu_strerror(error);
+            lu_error_free(&error);
             rv = false;
         }
+        g_slice_free(GValue, value);
```

Comment:
The memory for value is released, but how about the content within?
Should there be g_value_unset(value) or something to undo resources allocated at g_value_set_string()?

Reply:
`GValue`s are not big (24 bytes on most platforms), they can be (and often are) allocated on stack. `g_value_unset()` is needed in any case, though.

Reply:
Unset added